src/GitHub.VisualStudio/Menus/BlameLink.cs

```diff
@@ -0,0 +1,42 @@
+﻿using GitHub.Services;
+using GitHub.VisualStudio.UI;
+using NullGuard;
+using System;
+using System.Windows;
+
+namespace GitHub.VisualStudio.Menus
+{
+    public class BlameLink : LinkMenuBase, IDynamicMenuHandler
+    {
+
+        public BlameLink(IGitHubServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+        }
+
+        public Guid Guid => GuidList.guidContextMenuSet;
+        public int CmdId => PkgCmdIDList.blameCommand;
+
+        public async void Activate([AllowNull]object data = null)
+        {
+            var isgithub = await IsGitHubRepo();
+            if (!isgithub)
+                return;
+
+            var link = await GenerateLink(true);
```

Comment:
No idea why I went against the named param and removed it. Glad I was on the same track, though!
I added the optional bool to avoid altering the Create/Copy commands. Enum sounds better in the event we start adding more urls, so I think I'll go with that.